docs/reference/builder.md

```diff
@@ -1584,20 +1584,22 @@ If you forget to add `exec` to the beginning of your `ENTRYPOINT`:
 ```dockerfile
 FROM ubuntu
 ENTRYPOINT top -b
-CMD --ignored-param1
+CMD "--ignored-param1"
```

Comment:
Wondering if this is just "tricking" the parser to consider it valid (by adding the quotes), and if instead it should be

```suggestion
CMD -- --ignored-param1
```

(using `--` to terminate processing as option for `CMD`)

However, in both cases, this likely won't give the expected result, as `CMD` will still become `["/bin/sh", "-c", ...]`, so it's not really ignored

```dockerfile
FROM docker
ENTRYPOINT docker run
CMD "--help"
```

```bash
docker inspect --format='{{json .Config.Entrypoint }} {{json .Config.Cmd }}' foo
["/bin/sh","-c","docker run"] ["/bin/sh","-c","\"--help\""]
```

```dockerfile
FROM docker
ENTRYPOINT docker run
CMD -- --help
```

```bash
docker inspect --format='{{json .Config.Entrypoint }} {{json .Config.Cmd }}' foo
["/bin/sh","-c","docker run"] ["/bin/sh","-c","--help"]
```


I'm trying to understand what the purpose of the example is, because the `--ignored-param` (unless there's a binary with that name) would never give anything useful 🤔

Reply:
Thanks, the `CMD -- --xxx` looks much better.

Well the example shows the edge cases and it seems both ENTRYPOINT and CMD without using an array doesn't build a useful command, it always has a second `/bin/sh` 🤷